packages/acceptit-Core.package/ACScenarioSemantic.class/instance/scenario.withGiven.andLinebreak.andAndLinebreak.andWhen.andLinebreak.andAndLinebreak.andThen.andLinebreakAnd..st

```diff
@@ -7,18 +7,23 @@ andAndLinebreak: and2
 andThen: aThenRule  
 andLinebreakAnd: and3
 
-	|return andResult1 andResult2 andResult3|
-	
+	|return givens whens thens andResult1 andResult2 andResult3|
 	return := OrderedCollection new.
-	return add: (self value: aGivenRule).
+	givens := OrderedCollection new.
+	givens add: (self value: aGivenRule).
 	self value: linebreak1.
 	andResult1 := self value: and1.
-	(andResult1 isNil) ifFalse: [return add: andResult1].
-	return add: (self value: aWhenRule).
+	andResult1 do: [:each| givens add: each].
```

Comment:
not really sure why it is still necessary to differentiate between statement and the respective and statements. Could they be entered into the method all as givens? At the end they are added all to the local var givens with no differentiation anyway

Reply:
See new method for explanation. its cleaner now